### ISSUE TITLE
Haskell daemons exit on SIGHUP when running in the foreground

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#811]
Originally reported of Google Code with ID 755.

```
<b>What steps will reproduce the problem?</b>
1. Spawn a Haskell daemon (e.g. mond) in the foreground
2. Send the daemon the HUP signal
3. The daemon exits

The Ganeti Haskell daemon implementation adds the SIGHUP handler only on daemonization. Foreground processes never get a chance to initialize their SIGHUP handler and when the signal is delivered, the process carries on the default action which is to exit. The expected behavior would be for the daemons to rotate their logfiles as normal. This is essential for running the daemons in supervised mode (e.g. using systemd).

It should be noted that the Python daemons correctly set up the SIGHUP handlers when run in the foreground.
```
Originally added on 2014-03-10 11:52:14 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2014-03-10 11:52:37 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2014-03-10 13:49:26 +0000 UTC.

Changed State: Started

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2014-05-13 13:52:01 +0000 UTC.

Changed State: Accepted

**Comment by Ganeti-Issues-Migrator** [comment]
```
Fixed in:

commit fef5c09ec3c0595f90889957e424cbd08cff5fc4
Author: Apollon Oikonomopoulos <apoikos@gmail.com>
Date:   Mon Sep 29 17:41:38 2014 +0300

    Ganeti.Daemon: always install SIGHUP handler
    
    Install the SIGHUP handler regardless of the daemonization status. This
    fixes issue #755.
    
    Signed-off-by: Apollon Oikonomopoulos <apoikos@gmail.com>
    Signed-off-by: Petr Pudlak <pudlak@google.com>
    Reviewed-by: Petr Pudlak <pudlak@google.com>

```
Originally added on 2014-10-06 14:46:23 +0000 UTC.

Changed State: Fixed

**Comment by Ganeti-Issues-Migrator** [comment]
```
Released as part of 2.14.0
```
Originally added on 2015-06-02 10:41:44 +0000 UTC.

Changed State: Released
